rpctest/rpc_harness_test.go

```diff
@@ -0,0 +1,454 @@
+		addr, err := r.NewAddress()
+		if err != nil {
+			t.Fatalf("unable to get new address: %v", err)
+		}
+
+		// Next, send amt BTC to this address, spending from one of our mature
+		// coinbase outputs.
+		addrScript, err := txscript.PayToAddrScript(addr)
+		if err != nil {
+			t.Fatalf("unable to generate pkscript to addr: %v", err)
+		}
+		output := wire.NewTxOut(int64(amt), addrScript)
+		txid, err := r.SendOutputs([]*wire.TxOut{output}, 10)
+		if err != nil {
+			t.Fatalf("coinbase spend failed: %v", err)
+		}
+		return txid
+	}
+
+	// First, generate a small spend which will require only a single
+	// input.
+	txid := genSpend(btcutil.Amount(5 * btcutil.SatoshiPerBitcoin))
+
+	// Generate a single block, the transaction the wallet created should
+	// be found in this block.
+	blockHashes, err := r.Node.Generate(1)
+	if err != nil {
+		t.Fatalf("unable to generate single block: %v", err)
+	}
+
+	block, err := r.Node.GetBlock(blockHashes[0])
+	if err != nil {
+		t.Fatalf("unable to get block: %v", err)
+	}
+
+	minedTx := block.Transactions()[1]
+	txSha := minedTx.Hash()
+	if !bytes.Equal(txid[:], txSha[:]) {
+		t.Fatalf("txid's don't match, %v vs %v", txSha, txid)
+	}
+
+	// Next, generate a spend much greater than the block reward.
+	genSpend(btcutil.Amount(500 * btcutil.SatoshiPerBitcoin))
+}
+
+func assertConnectedTo(t *testing.T, nodeA *Harness, nodeB *Harness) {
+	nodePort := defaultP2pPort + (2 * nodeB.nodeNum)
+	nodeAddr := net.JoinHostPort("127.0.0.1", strconv.Itoa(nodePort))
+
+	nodeAPeers, err := nodeA.Node.GetPeerInfo()
+	if err != nil {
+		t.Fatalf("unable to get harness1's peer info")
+	}
+
+	addrFound := false
+	for _, peerInfo := range nodeAPeers {
+		if peerInfo.Addr == nodeAddr {
+			addrFound = true
+			break
+		}
+	}
+
+	if !addrFound {
+		t.Fatal("nodeA not connected to nodeB")
+	}
+}
+
+func testConnectNode(r *Harness, t *testing.T) {
+	// Create a fresh test harnesses.
+	harness, err := New(&chaincfg.SimNetParams, nil, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := harness.SetUp(true, 25); err != nil {
+		t.Fatalf("unable to complete rpctest setup: %v", err)
+	}
+	defer harness.TearDown()
+
+	// Establish a p2p connection the main harness to our new local
+	// harness.
+	if err := ConnectNode(r, harness); err != nil {
+		t.Fatalf("unable to connect harness1 to harness2: %v", err)
+	}
+
+	// The main harness should show up in our loca harness' peer's list,
+	// and vice verse.
+	assertConnectedTo(t, r, harness)
+}
+
+func testTearDownAll(t *testing.T) {
+	// Tear down all currently active harnesses.
+	if err := TearDownAll(); err != nil {
+		t.Fatalf("unable to teardown all harnesses: %v", err)
+	}
+
+	// The global testInstances map should now be fully purged with no
+	// active test harnesses remaining.
+	activeHarnesses := ActiveHarnesses()
+	if len(activeHarnesses) != 0 {
+		t.Fatalf("test harnesses still active after TestTearDownAll()")
+	}
+
+	for _, harness := range activeHarnesses {
```

Comment:
This was just checked for `!= 0` above and the test fails if not, so this will not do anything as there is nothing to range.  I think you meant to get a local copy before calling `TearDownAll` above and use that.